src/robot/libraries/BuiltIn.py

```diff
@@ -2232,7 +2244,16 @@ def wait_until_keyword_succeeds(self, retry, retry_interval, name, *args):
                     raise AssertionError("Keyword '%s' failed after retrying "
                                          "%s. The last error was: %s"
                                          % (name, message, err))
-                self._sleep_in_parts(retry_interval)
+                keyword_runtime = time.time() - start_time
+                if strict_interval:
+                    sleep_time = retry_interval - keyword_runtime
+                if sleep_time < 0:
```

Comment:
It would be a bit more clear that this `if` can be true only if the above `if` is true, if it would be inside the above `if`.